src/engine/renderer/tr_shader.cpp

```diff
@@ -1479,6 +1482,58 @@ static bool ParseClampType( char *token, wrapType_t *clamp )
 	return true;
 }
 
+/*
+===================
+FindMapInStage
+===================
+*/
+static bool FindMapInStage( const char **text, char *buffer, int bufferlen )
+{
+	char *token;
+	bool foundMap;
+
+	while ( true )
```

Comment:
This function could use some comments explaining what it's looking for or what it's trying to do. Is this the exactly algorithm of Darkplaces or is it some kind of approximation? An example like the ones in the PR comments would be good.

Reply:
I will add a comment, this part of the code has no chance to be the same algorithm of Darkplaces, the engines are too much different on this part, there is only high level ideas like "for basename look for suffixes” to share. The only part that is probably shareable with Darkplaces are the suffixes themselves…